examples/vector_store_search.yaml

```diff
@@ -1,28 +1,40 @@
 ---
-# Example that uses Cohere embeddings and OpenSearch for vector store 
+# Example that uses Cohere embeddings and Amazon OpenSearch Service Serverless for vector store 
 # This also shows how to use AWS credentials and AWS4Auth for OpenSearch
 # which involves using 'invoke' to create the required auth objects
+# 
+# 
+# Follow Boto3 documentation for AWS credentials: 
+# https://boto3.amazonaws.com/v1/documentation/api/latest/guide/quickstart.html#configuration
+# https://python.langchain.com/v0.2/docs/integrations/vectorstores/opensearch/#using-aoss-amazon-opensearch-service-serverless
+#
+# Dependencies:
+# pip install -U langchain_community opensearch-py requests_aws4auth
+#
+# required ENV variables:
+# - AWS_BEDROCK_COHERE_EMBED_MODEL_ID
+# - AWS_BEDROCK_COHERE_EMBED_REGION
+# - AWS_OPENSEARCH_INDEX_NAME
+# - AWS_OPENSEARCH_ENDPOINT
+
 log:
   stdout_log_level: DEBUG
   log_file_level: DEBUG
   log_file: solace_ai_connector.log
 
 shared_config:
-  - broker_config: &broker_connection
-      broker_connection_share: ${SOLACE_BROKER_URL}
-      broker_type: solace
-      broker_url: ${SOLACE_BROKER_URL}
-      broker_username: ${SOLACE_BROKER_USERNAME}
-      broker_password: ${SOLACE_BROKER_PASSWORD}
-      broker_vpn: ${SOLACE_BROKER_VPN}
-
-  # Get AWS credentials object
+  # Get AWS credentials object from .aws credentials
+  # You can pass the ACCESS/SECRET/SESSION keys directly as ENV variables as well
+  # eg: aws_secret_access_key: ${AWS_SECRET_ACCESS_KEY}
   - aws_credentials: &aws_credentials
       invoke:
         object:
           invoke:
             module: boto3
             function: Session
+            params:
+              keyword:
+                profile_name: default # The profile to choose from .aws/credentials
```

Comment:
I guess this was the secret sauce to get this working again

Reply:
Actually, the main issues were from AWS configurations side